src/main/java/com/yelp/nrtsearch/server/luceneserver/field/TermQueryableIndexableFieldDef.java

```
@@ -0,0 +1,154 @@
+import com.yelp.nrtsearch.server.luceneserver.field.properties.TermQueryable;
+import org.apache.lucene.search.Query;
+
+public abstract class TermQueryableIndexableFieldDef extends IndexableFieldDef
+    implements TermQueryable {
+  /**
+   * Field constructor. Performs generalized building of field definition by calling a number of
+   * protected methods, mainly: {@link #validateRequest(Field)}, {@link #parseDocValuesType(Field)},
+   * {@link #parseFacetValueType(Field)}, and {@link #setSearchProperties(FieldType, Field)}.
+   * Concrete field types should override these methods as needed.
+   *
+   * @param name name of field
+   * @param requestField field definition from grpc request
+   */
+  protected TermQueryableIndexableFieldDef(String name, Field requestField) {
+    super(name, requestField);
+  }
+
+  // Performs type validation. Do not @Override in subclasses.
+  @Override
+  public Query getTermQuery(TermQuery termQuery) {
+    switch (termQuery.getTermTypesCase()) {
+      case BOOLEANVALUE:
+        return getTermQueryFromBooleanValue(termQuery);
+      case DOUBLEVALUE:
+        return getTermQueryFromDoubleValue(termQuery);
+      case FLOATVALUE:
+        return getTermQueryFromFloatValue(termQuery);
+      case INTVALUE:
+        return getTermQueryFromIntValue(termQuery);
+      case LONGVALUE:
+        return getTermQueryFromLongValue(termQuery);
+      case TEXTVALUE:
+        return getTermQueryFromTextValue(termQuery);
+      default:
+        throw new IllegalArgumentException(
+            String.format(
+                "%s field does not support term type: %s",
+                termQuery.getField(), termQuery.getTermTypesCase()));
+    }
+  }
+
+  Query getTermQueryFromBooleanValue(TermQuery termQuery) {
+    throwIllegalArgumentException(termQuery);
+    return null;
+  }
+
+  Query getTermQueryFromDoubleValue(TermQuery termQuery) {
+    throwIllegalArgumentException(termQuery);
+    return null;
```

It might be better to throw an UnsupportedOperationException, or an IllegalStateException. The other option would be for these methods to return null, and do the exception construction in the main query building method.